easybuild/tools/options.py

```diff
@@ -724,6 +729,30 @@ def postprocess(self):
             self.show_config()
             cleanup_and_exit(self.tmpdir)
 
+    def _postprocess_optarch(self):
+        """Postprocess --optarch option."""
+        optarch_parts = self.options.optarch.split(OPTARCH_SEP)
+        
+        # we expect to find a ':' in every entry in optarch, in case optarch is specified on a per-compiler basis
+        if (len(optarch_parts) > 1 and not all(p.count(OPTARCH_MAP_CHAR) == 1 for p in optarch_parts)) or \
+                (len(optarch_parts) == 1 and optarch_parts[0].count(OPTARCH_MAP_CHAR) > 1):
```

Comment:
Looks more messy to me..... The `if` is shorter, sure, but everything else seems unnecessarly complicated to me. But I don't really care....